recipe/meta.yaml

```diff
@@ -45,11 +45,8 @@ requirements:
     - openblas
   run:
     - python
-    - {{ pin_compatible('numpy') }}
```

Comment:
numpy has switched to using run exports (and it has been for 2 years)

xref: [numpy2 migration announcement](https://conda-forge.org/news/2024/05/15/numpy-2-migration/), https://github.com/conda-forge/tiledb-py-feedstock/pull/218